headsplugin-api/src/main/java/com/github/cc007/headsplugin/api/business/services/heads/CategorySearcher.java

```diff
@@ -0,0 +1,32 @@
+package com.github.cc007.headsplugin.api.business.services.heads;
+
+import com.github.cc007.headsplugin.api.business.domain.Category;
+import com.github.cc007.headsplugin.api.business.domain.Head;
+
+import java.util.Set;
+
+public interface CategorySearcher {
```

Comment:
Searching should be redone with repositories IMO

Reply:
Repositories are not public facing constructs. They are part of the integration layer and should only be used by the service layer. This is how a service oriented architecture (SOA) works.